### PR TITLE
Add env vars to eks-operator for proxy support

### PR DESCRIPTION
**Pull request by rmweir**
**Problem:**
Eks-operator is not working in airgap proxy setups.

**Solution:**
Add fields for configuring proxy environment variables in deployment.

**Issue:**
https://github.com/rancher/rancher/issues/29070

### CODE REVIEW

**Comment by rmweir** [comment]
The only difference is in the deployment yaml. HTTPS_PROXY, HTTP_PROXY, NO_PROXY are now passed as env variables and are configurable through values.

File: charts/rancher-eks-operator/1.0.101/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+You have deployed the Rancher EKS operator
+Version: {{ .Chart.AppVersion }}
+Description: This operator provisions EKS clusters
+from EKSClusterConfig CRs.
```
Comment (nickgerace) [inline charts/rancher-eks-operator/1.0.101/templates/NOTES.txt:4]: Perhaps there's a rule I don't know; is this line wrapping intentional?

File: charts/rancher-eks-operator/1.0.101/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  template:
+    metadata:
+      labels:
+        ke.cattle.io/operator: eks
+    spec:
+      serviceAccountName: eks-operator
+      containers:
+      - name: eks-operator
+        image: {{ template "system_default_registry" . }}{{ .Values.eksOperator.image.repository }}:{{ .Values.eksOperator.image.tag }}
+        imagePullPolicy: IfNotPresent
```
Comment (nickgerace) [inline charts/rancher-eks-operator/1.0.101/templates/deployment.yaml:20]: Would there be an advantage to having the pull policy in ```values.yaml``` instead?

**Review by nickgerace — DISMISSED** [review]

File: charts/rancher-eks-operator/1.0.101/templates/deployment.yaml (hunk 2 above)
Comment (rmweir) [reply charts/rancher-eks-operator/1.0.101/templates/deployment.yaml:20]: Probably not, we have no need to configure it at the moment.

File: charts/rancher-eks-operator/1.0.101/templates/NOTES.txt (hunk 1 above)
Comment (rmweir) [reply charts/rancher-eks-operator/1.0.101/templates/NOTES.txt:4]: Probably to be compatible with default printer column widths for commands such as `helm search repo`: https://helm.sh/docs/helm/helm_search_repo/#options

**Review by mrajashree — DISMISSED** [review]

**Comment by cbron** [comment]
For adding system-chart versions, the first commit should always be a copy of the existing chart, and the 2nd commit should be the changes involved, that way we can review changes easily. Ex: https://github.com/rancher/system-charts/pull/265/commits

**Comment by rmweir** [comment]
@cbron oh that definitely simplifies it. Updated.

File: charts/rancher-eks-operator/1.0.101/templates/deployment.yaml [hunk 3]
```diff
@@ -18,3 +18,10 @@ spec:
       - name: eks-operator
         image: {{ template "system_default_registry" . }}{{ .Values.eksOperator.image.repository }}:{{ .Values.eksOperator.image.tag }}
         imagePullPolicy: IfNotPresent
+        env:
+        - name: HTTP_PROXY
+          value: {{ .Values.httpProxy }}
```
Comment (cbron) [inline charts/rancher-eks-operator/1.0.101/templates/deployment.yaml:23]: Nit: Seems that it is ok if these are set to blank, but you could wrap in conditional so they aren't set at all if blank.

**Review by cbron — APPROVED** [review]
Questions min/max should be fine as-is.

File: charts/rancher-eks-operator/1.0.101/templates/deployment.yaml (hunk 3 above)
Comment (rmweir) [reply charts/rancher-eks-operator/1.0.101/templates/deployment.yaml:23]: The rancher code marshal the struct into yaml and passes it to helm without checking if each variable is empty. Therefore, they will never be empty as rancher uses it. I can do the check though.

Comment (cbron) [reply charts/rancher-eks-operator/1.0.101/templates/deployment.yaml:23]: Na its fine

**Pull request merged**